torchmetrics/detection/map.py

```diff
@@ -214,17 +194,28 @@ def __init__(
             dist_sync_fn=dist_sync_fn,
         )
 
-        if not _PYCOCOTOOLS_AVAILABLE:
-            raise ImportError(
-                "`MAP` metric requires that `pycocotools` installed."
-                " Please install with `pip install pycocotools` or `pip install torchmetrics[detection]`"
-            )
         if not (_TORCHVISION_AVAILABLE and _TORCHVISION_GREATER_EQUAL_0_8):
             raise ImportError(
                 "`MAP` metric requires that `torchvision` version 0.8.0 or newer is installed."
                 " Please install with `pip install torchvision` or `pip install torchmetrics[detection]`"
             )
 
+        self.box_format = box_format
```

Comment:
please add input validation
```suggestion
        allowed_box_formats = (‘xyxy’, ‘xywh’, ‘cxcywh’)
        if box_format not in allowed_box_formats:
            raise ValueError(
              f"Expected argument `box_format` to be one of {allowed_box_formats} but got {box_format}"
            )
        self.box_format = box_format
```